### DISCUSSION TITLE
Laser is not working

### DISCUSSION THREAD

**Discussion by MikeFreeroam** [arkypita/LaserGRBL#1105] [Q&A]
Please Can you help me I build my first DIY Laser engraver... I have arduino uno with cnc shield 3.0 and I soldered everything like is in the scheme...but when I soldered ground from laser to the Gate of (mosfet) laser small beam was there and when I soldered to the Souce (mosfet) laser was on full power all of the time but when I soldered to the Drain (mosfet) laser is not working at all...I use LaserGRBL and I tried every settings 30=256,32=1... Please help me and sorry for my english :D 
![IMG_0061](https://user-images.githubusercontent.com/70109677/90986340-69195b00-e582-11ea-9cdd-18046609a40f.JPG)
![IMG_0062](https://user-images.githubusercontent.com/70109677/90986344-71719600-e582-11ea-84ed-3803f0701b8a.JPG)
![laser648](https://user-images.githubusercontent.com/70109677/90986350-7e8e8500-e582-11ea-846c-8025ef1a53ce.jpg)

**Comment by StuartB4** [comment]
You can't use the SpnEn pin. If you have grbl 1.1 you need to use the Z+ or Z- pin instead. That is connected to the PWM pin 11 on the Arduino UNO.

**Comment by MikeFreeroam** [comment]
Ok thanks that works for a little bit laser beam was there but now I have these problems
![Bez názvu](https://user-images.githubusercontent.com/70109677/91056580-7425c700-e626-11ea-9e1f-91d084434212.png)

**Comment by MikeFreeroam** [comment]
Program now is lagging like crazy...

**Comment by StuartB4** [comment]
If it's not a genuine arduino you could try reinstalling the CH340 driver on the Tools menu.
Is your power supply strong enough? is it running the laser and the motors? 
Can you post a picture of your grbl configuration settings.

**Comment by MikeFreeroam** [comment]
Yes I can but I tried BENBOX and in benbox it works perfectly...of course I connected pin back to spindleEn...

**Comment by StuartB4** [comment]
Ok.

**Comment by MikeFreeroam** [comment]
But anyway thanks for your help...
